daemon/container_operations_unix.go

```diff
@@ -69,14 +69,14 @@ func (daemon *Daemon) getPidContainer(container *container.Container) (*containe
 	containerID := container.HostConfig.PidMode.Container()
 	container, err := daemon.GetContainer(containerID)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrapf(err, "cannot join PID of a non running container: %s", container.ID)
 	}
 	return container, daemon.checkContainer(container, containerIsRunning, containerIsNotRestarting)
 }
 
 func containerIsRunning(c *container.Container) error {
 	if !c.IsRunning() {
-		return fmt.Errorf("cannot join IPC of a non running container: %s", c.ID)
+		return errors.Errorf("container %s is not running", c.ID)
```

Comment:
Actually, looks like these changes also belong in 25caa5e9194db2cee946e9ef98557dca9976f6d3

https://github.com/docker/docker/pull/30843/commits/25caa5e9194db2cee946e9ef98557dca9976f6d3#r102823906